src/SonataPageAdminTrait.php

```diff
@@ -0,0 +1,312 @@
+        $button        = $this->getSession()->getPage()->find('css', $buttonElement);
+
+        if (null === $button) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'button', $button);
+        }
+
+        $this->getSession()->executeScript(sprintf('window.scrollTo($(\'%s\').position().top+200, $(\'%s\').position().left+200);', $buttonElement, $buttonElement));
+
+        $button->press();
+
+        $this->waitForSeconds(1);
+
+        $button = $this->getSession()->getPage()->find('css', 'button.btn-danger');
+
+        if (null === $button) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'button', $button);
+        }
+
+        $button->click();
+    }
+
+    /**
+     * I rename the block with.
+     *
+     * @When  /^I rename the block "([^"]*)" with "([^"]*)"$/
+     *
+     * @param string $blockName
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iRenameTheBlock(string $blockName, string $name): void
+    {
+        $this->iOpenTheBlock($blockName);
+
+        $block = $this->getSession()->getPage()->find('css', sprintf('li:contains(\'%s\')', $blockName));
+
+        if (null === $block) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $block);
+        }
+
+        $id      = $block->getAttribute('data-block-id');
+        $element = sprintf('li[data-block-id=%d] * input.page-composer__container__child__name__input', $id);
+
+        $this->iWaitForCssElementBeingVisible($element, 2);
+
+        $input = $this->getSession()->getPage()->find('css', $element);
+
+        if (null === $input) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'input', $block);
+        }
+
+        $input->setValue($name);
+
+        $button = $this->getSession()->getPage()->find('css', sprintf('li[data-block-id=%d] * button[type=submit]', $id));
+
+        if (null === $button) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $button);
+        }
+
+        $button->press();
+    }
+
+    /**
+     * The block should be opened.
+     *
+     * @When /^The block "([^"]*)" should be opened$/
+     *
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function theBlockShouldBeOpened(string $name): void
+    {
+        $block = $this->getSession()->getPage()->find('css', sprintf('li:contains(\'%s\')', $name));
+
+        if (null === $block) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'block', $block);
+        }
+
+        $id      = $block->getAttribute('data-block-id');
+        $element = sprintf('li.page-composer__container__child[data-block-id=%d] > div.page-composer__container__child__content', $id);
+
+        $this->waitForSeconds(3);
```

Comment:
3 seconds seems very long as it will always wait 3 seconds before doing anything. What 's the matter behind this?

Reply:
why resolved?

Reply:
why resolved? can you explain please?

Reply:
sorry not resolved, I'll investigate

Reply:
we can delete this `waitForSeconds` function